### PR TITLE
Update llama_index to 0.9.3

### PR DESCRIPTION
**Pull request by lopagela**
Update llama_index to 0.9.3

Had to change some imports because of breaking change during the llama_index update to 0.9.0

Finally drop `langchain` and langsmith from the dependencies (making the installation lighter and faster)

----

llama_index 0.9.0 announcement: https://blog.llamaindex.ai/announcing-llamaindex-0-9-719f03282945

Changelogs of llama_index: https://github.com/run-llama/llama_index/blob/main/CHANGELOG.md

### CODE REVIEW

File: poetry.lock
Comment (lopagela) [inline poetry.lock:1]: Finally dropping langchain and langsmith in this file

File: private_gpt/server/ingest/ingest_service.py [hunk 1]
```diff
@@ -6,14 +6,13 @@
...
     ServiceContext,
     StorageContext,
-    StringIterableReader,
     VectorStoreIndex,
     load_index_from_storage,
 )
 from llama_index.node_parser import SentenceWindowNodeParser
+from llama_index.readers import JSONReader, StringIterableReader
 from llama_index.readers.file.base import DEFAULT_FILE_READER_CLS
 from pydantic import BaseModel, Field
```
Comment (lopagela) [inline private_gpt/server/ingest/ingest_service.py:17]: Some changes in how the imports are structured in llama_index

File: pyproject.toml [hunk 2]
```diff
@@ -12,7 +12,7 @@ injector = "^0.21.0"
 pyyaml = "^6.0.1"
 python-multipart = "^0.0.6"
 pypdf = "^3.16.2"
-llama-index = "0.8.67"
+llama-index = { extras = ["local_models"], version = "0.9.3" }
```
Comment (lopagela) [inline pyproject.toml:15]: Install `llama-index` to run models locally

**Review by imartinez — APPROVED** [review]
Amazing!

**Pull request merged**